build-llvm.py

```diff
@@ -385,32 +414,42 @@ def cc_ld_cmake_defines(dirs, env_vars, stage):
     if stage == 1:
         ar = None
         cc = env_vars.cc
+        clang_tblgen = None
         cxx = env_vars.cxx
         ld = env_vars.ld
+        llvm_tblgen = None
         ranlib = None
     else:
         ar = get_stage1_binary("llvm-ar", dirs)
         cc = get_stage1_binary("clang", dirs)
+        clang_tblgen = get_stage1_binary("clang-tblgen", dirs)
         cxx = get_stage1_binary("clang++", dirs)
         ld = get_stage1_binary("ld.lld", dirs)
+        llvm_tblgen = get_stage1_binary("llvm-tblgen", dirs)
         ranlib = get_stage1_binary("llvm-ranlib", dirs)
 
-    # Use llvm-ar for stage 2 builds to avoid errors with bfd plugin
+    # Use llvm-ar for stage 2+ builds to avoid errors with bfd plugin
     # bfd plugin: LLVM gold plugin has failed to create LTO module: Unknown attribute kind (60) (Producer: 'LLVM9.0.0svn' Reader: 'LLVM 8.0.0')
     if ar is not None:
         defines['CMAKE_AR'] = ar
 
     # The C compiler to use
     defines['CMAKE_C_COMPILER'] = cc
 
+    if clang_tblgen is not None:
+        defines['CLANG_TABLEGEN'] = clang_tblgen
+
     # The C++ compiler to use
     defines['CMAKE_CXX_COMPILER'] = cxx
 
     # If we have a linker, use it
     if ld is not None:
         defines['LLVM_USE_LINKER'] = ld
 
-    # Use llvm-ranlib for stage 2 builds
+    if llvm_tblgen is not None:
```

Comment:
```suggestion
    if llvm_tblgen:
```

Reply:
```
Python 3.6.8 (default, Jan 14 2019, 11:02:34) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> llvm_tblgen = None
>>> if llvm_tblgen:
...     print("llvm_tblgen is not None")
... else:
...     print("llvm_tblgen is None")
... 
llvm_tblgen is None
```

Reply:
Will do that for all of the if statements.